### ISSUE TITLE
Bluetooth Connectivity with RealDash

### ISSUE THREAD

**Issue by martinmain** [meatpiHQ/wican-fw#63]
Hi I seem to be having a problem with Bluetooth connectivity between my Samsung Galaxy S8 Android device which is running RealDash and the MeatPi OBD2 device v1.80 firmware v3.00 hardware. I am operating in realdash66 mode to access canbus data and not in elm327 mode. It seems to make initial connection, requests the passcode and I can even see a burst of canbus data in the RealDash CAN Monitor then it asks for the passcode again and again and again. Any ideas what might be going on?

**Comment by martinmain** [comment]
Many thanks for prompt reply, I will give this a try.

**Comment by martinmain** [comment]
Hi i tried connecting and although I did manage to get it to connect it would not bond despite several attempts, I am attaching the log file from nrf connect
[Log 2023-08-21 18_13_32.txt](https://github.com/meatpiHQ/wican-fw/files/12397089/Log.2023-08-21.18_13_32.txt)

**Comment by meatpiHQ** [comment]
@martinmain from the logs I can see it's connected and receiving data. The frames that start with 0x66 are CAN bus frames. Did you try RealDash again?

**Comment by martinmain** [comment]
I followed your advice and tried again after using nrf connect and whilst I had some success with parameters updating on RealDash this was short lived and it stopped communicating on BLE again but I had it operating for circa 5 mins perfectly. I have reverted back to AP Station mode, but perhaps should have mentioned earlier WiFi is also dropping out intermittently. I have no problems getting WiFi connectivity but periodically while I am driving (maybe once or sometimes twice in 30 mins) it loses the connection and re-establishes itself. Possible hardware problem?

**Comment by meatpiHQ** [comment]
@martinmain are you using realdash on iOS?

**Comment by martinmain** [comment]
No it’s a Samsung S8 running Android

**Comment by martinmain** [comment]
Yes, I have sleep mode set at 12.8V with car sitting at 13.4V while running

Pics as attached

![two](https://github.com/meatpiHQ/wican-fw/assets/29522586/c3bd765c-daec-40fc-9efa-2e631ab755f8)
![one](https://github.com/meatpiHQ/wican-fw/assets/29522586/7f8f0b80-6b1d-4bd4-b922-3d95bd3363f3)
![three](https://github.com/meatpiHQ/wican-fw/assets/29522586/b82c3641-c4ad-432b-a457-3de944bc69e1)

**Comment by meatpiHQ** [comment]
@martinmain try to set it to AP mode instead of AP+Station mode.

**Comment by martinmain** [comment]
The realdash device needs internet access via WiFi in this mode it will only be connected to the MeatPi device?

Sent from my iPhone

On 22 Aug 2023, at 19:07, meatpiHQ ***@***.***> wrote:

﻿


@martinmain try to set it to AP mode instead of AP+Station mode.

**Comment by meatpiHQ** [comment]
@martinmain RealDash can work without Internet connection. This is only temporary it will help to find the issue.

**Comment by martinmain** [comment]
Do you think a clean install of the Android device is worth considering, since I assume BLE should work on Android without issue?

**Comment by meatpiHQ** [comment]
@martinmain I don't think it will help. But there are some settings in android related to battery optimization might be causing an issue. 

I also have a couple of ideas what might be causing this in the firmware, I'll need to do some testing, I might send you a new firmware to try tomorrow.

Please try and revert back to v1.71 and try again with the BLE.

https://github.com/meatpiHQ/wican-fw/releases/tag/v1.71

**Comment by martinmain** [comment]
Many thanks for your support, I will give this a try later today and let you know how I get on

**Comment by martinmain** [comment]
I downgraded the firmware to 1.71 as suggested, but unfortunately it is not any better, it connects momentarily before asking for passcode again repeatedly. Although probably not relevant WiFi is no better either 2 km of my 28 km commute were missed due it reconnecting multiple times.

**Comment by martinmain** [comment]
Hi loaded v2.00 as per your guidance and pleased to report BLE connectivity seems to be working fine I was getting circa 750 frames / second in RealDash CAN Monitor. I only tested for 10-15 minutes but observed no data dropouts. Assume the battery sleep function is disabled in firmware?

**Comment by meatpiHQ** [comment]
@martinmain Sleep is not disabled, but it's not tested yet. I migrated to IDF5 and I think it's broken because there's some changes to ADC api. I'm still working on it, I'll release a beta version tomorrow.

**Comment by martinmain** [comment]
Noted and many thanks for your support

**Comment by meatpiHQ** [comment]
@martinmain I released the new [v2.10 beta version.](https://github.com/meatpiHQ/wican-fw/releases/tag/v2.10)

**Comment by martinmain** [comment]
Pleased to report BLE is rock solid, have not had the opportunity to try WiFi yet. One thing I did notice and not sure if it's the device or RealDash, but when I turn off the ignition, values are held at their last running value I assume until the device goes to sleep. For most this is okay, but RPM sits at idle speed with the engine off.

**Comment by martinmain** [comment]
Is there any possibility of getting the 180 second sleep timer available as a user configurable option?

**Comment by martinmain** [comment]
@meatpiHQ just wondering if the recent 2,50 revision contains the same BLE and WiFi mods as 2.10beta?

**Comment by meatpiHQ** [comment]
@martinmain Yes, the latest version has some additional features and improvements to MQTT.

**Comment by martinmain** [comment]
@meatpiHQ installed 2.50 and was unable to get BLE working, so reinstalled 2.10 beta
